onprc_ehr/resources/web/onprc_ehr/form/field/ProjectEntryField.js

```diff
@@ -0,0 +1,364 @@
+            this.triggerEl.set({'data-qtip': 'Click to recalculate allowable projects'});
+        }, this);
+    },
+
+    getInnerTpl: function(){
+        return ['<span style="white-space:nowrap;{[values["isAssigned"] ? "font-weight:bold;" : ""]}">{[LABKEY.Utils.encodeHtml(values["displayName"] + " " + (values["shortname"] ? ("(" + values["shortname"] + ")") : (values["investigator"] ? "(" + (values["investigator"] ? values["investigator"] : "") : "") + (values["account"] ? ": " + values["account"] : "") + (values["investigator"] ? ")" : "")))]}&nbsp;</span>'];
+    },
+
+    trigger1Cls: 'x4-form-search-trigger',
+
+    onTrigger1Click: function(){
+        var boundRecord = EHR.DataEntryUtils.getBoundRecord(this);
+        if (!boundRecord){
+            Ext4.Msg.alert('Error', 'Unable to locate associated animal Id');
+            return;
+        }
+
+        var id = boundRecord.get('Id');
+        if (!id){
+            Ext4.Msg.alert('Error', 'No animal Id provided');
+            return;
+        }
+
+        this.getProjects(id);
+    },
+
+    getDisallowedProtocols: function(){
+        return null;
+    },
+
+    makeSql: function(id, date){
+        if (!id && !this.includeDefaultProjects)
+            return;
+
+        //avoid unnecessary reloading
+        var key = id + '||' + date;
+        if (this.loadedKey == key){
+            return;
+        }
+        this.loadedKey = key;
+
+        var sql = "SELECT DISTINCT t.project, t.displayName, t.account, t.protocolDisplayName, t.protocol, t.investigator, t.title, t.shortname, false as fromClient, min(sort_order) as sort_order, max(isAssigned) as isAssigned FROM (";
+
+        if (id){
+            //NOTE: show any actively assigned projects, or projects under the same protocol.  we also only show projects if either the animal is assigned, or that project is active
+            sql += "SELECT p.project as project, p.displayName as displayName, p.account as account, p.protocol.displayName as protocolDisplayName, " +
+                    " p.protocol as protocol, p.title, p.shortname, p.investigatorId." + this.invesLastNameCol + " as investigator, " +
+                    " CASE WHEN (a.project = p.project AND p.use_category = 'Research') THEN 0 WHEN (a.project = p.project) THEN 1 ELSE 2 END as sort_order, " +
+                    " CASE WHEN (a.project = p.project) THEN 1 ELSE 0 END as isAssigned " +
+                    " FROM ehr.project p JOIN study.assignment a ON (a.project.protocol = p.protocol) " +
+                    " WHERE a.id='"+id+"' AND (a.project = p.project) "; //TODO: restore this OR p.enddate IS NULL OR p.enddate >= curdate()
+
+            //NOTE: if the date is in the future, we assume active projects
+            if (date){
+                sql += "AND cast(a.date as date) <= '"+date.format('Y-m-d')+"' AND ((a.enddateCoalesced >= '"+date.format('Y-m-d')+"') OR ('"+date.format('Y-m-d')+"' >= now() and a.enddate IS NULL))";
```

Comment:
```suggestion
        sql += "AND cast(a.date as date) <= '"+Ext4.Date.format(date, 'Y-m-d')+"' AND ((a.enddateCoalesced >= '"+Ext4.Date.format(date, 'Y-m-d')+"') OR ('"+Ext4.Date.format(date, 'Y-m-d')+"' >= now() and a.enddate IS NULL))";
```

Reply:
The previous code allowed the Investigator name to be displayed, whereas the new coding format that your provided prevents the investigator name from being displayed.
This is the only reason I created a new onprc_ehr version of the same file name.

Is there an alternative code that you can suggest that allows the investigator name to be displayed?

Raymond

Reply:
Sorry, Josh. i left out some code.  The code you provided works very well as designed. I will resubmit the changes to the program again to you